slm_lab/agent/algorithm/actor_critic.py

```diff
@@ -314,15 +314,16 @@ def calc_gae_advs_v_targets(self, batch):
         Used for training with GAE
         '''
         v_preds = self.calc_v(batch['states'])
+        # v targets = r_t + gamma * V(s_(t+1))
+        next_v_preds = self.calc_v(batch['next_states'])
+        v_targets = math_util.calc_nstep_returns(batch, self.gamma, 1, next_v_preds)
         # calc next_state boundary value and concat with above for efficiency
-        next_v_pred_tail = self.calc_v(batch['next_states'][-1:])
+        next_v_pred_tail = next_v_preds[-1:]
         next_v_preds = torch.cat([v_preds[1:], next_v_pred_tail], dim=0)
```

Comment:
now this is just next_v_preds. can be removed. line 319-321 can be removed

Reply:
I see what this is doing now. :) Fixed.